tests/beta/test_default_qubit_tf.py

```diff
@@ -1011,9 +962,33 @@ def circuit(a):
         expected = tf.sin(a)
         assert np.allclose(grad, expected, atol=tol, rtol=0)
 
-    def test_tf_interface_gradient(self, tol):
-        """Tests that the gradient of the circuit fixture above is correct
-        using the TensorFlow interface"""
+    def test_prob_differentiability(self, tol):
+        """Test that the device probability can be differentiated"""
+        dev = qml.device("default.qubit.tf", wires=2)
+
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit(a, b):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.probs(wires=[1])
+
+        a = tf.Variable(0.54)
+        b = tf.Variable(0.12)
+
+        with tf.GradientTape() as tape:
+            res = circuit(a, b)[0]
+            res = res[1] - res[0]

```

Comment:
would be good to use clearer names here, found it hard to follow

Reply:
Updated to make it clearer:
```python
with tf.GradientTape() as tape:
    # get the probability of wire 1
    prob_wire_1 = circuit(a, b)[0]
    # compute Prob(|1>_1) - Prob(|0>_1)
    res = prob_wire_1[1] - prob_wire_1[0]
```